### DISCUSSION TITLE
Driver release schedule

### DISCUSSION THREAD

**Discussion by Rosentti** [NVIDIA/open-gpu-kernel-modules#374] [Q&A] [answered]
How often should we expect new NVIDIA drivers (open or closed-source)?

**Comment by mtijanic** [comment, marked as answer]
The exact dates depend on a lot of factors, but historically a _major_ driver release comes every 3-6 months:
![image](https://user-images.githubusercontent.com/7016472/190854816-485e843d-34d6-4df3-848c-c1d7b8378e47.png)

These can have up to a dozen minor releases afterwards, often overlapping with upcoming major releases, because some branches get long term support. These schedules change rapidly so we try to not make any promises regarding dates.

Additionally, sometimes features get pushed out from one major release onto the next, so even saying "Will be fixed in release 52x.xx" is not always possible.

FWIW, Open- and Closed-source drivers have the same release schedule, and the `.run` file actually contains the sources too.

**Reply by aaronp24** [reply]
Just to clarify one point about this answer: the .run package contains the contents of the `kernel-open/` directory and a prebuilt copy of `nv-kernel.o_binary` built from the contents of the `src/` directory, but it does _not_ include the contents of the `src/` directory itself. You need to fetch this repository to get those sources.
